src/actions/guides/database/models.cr

```diff
@@ -240,6 +252,66 @@ class Guides::Database::Models < GuideAction
 
     > The associations *must* be declared on both ends (the Post and the Tag in this example),
     > otherwise you will get a compile time error
+
+    ## Polymorphic
+
+    [Polymorphism](https://en.wikipedia.org/wiki/Polymorphism_(computer_science)) is describes
+    the concept that objects of different types can be accessed through the same interface.
+
+    This allows us to have a single method to define an association, but that method can return
+    many different types. A bit confusing, but best explained with some code!
+
+    ```crystal
+    class User < BaseModel
+      table do
+        has_many events : Event
+      end
+    end
+
+    class Organization < BaseModel
+      table do
+        has_many events : Event
+      end
+    end
+
+    class Event < BaseModel
+      table do
+        # Note that these are both nilable
+        belongs_to user : User?
+        belongs_to organization : Organization?
+
+        # Now `eventable` could be a `user` or `organization`
+        polymorphic eventable, associations: [:user, :organization]
+      end
+    end
+    ```
+
+    And to use it
+
+    ```crystal
+    user = SaveUser.create!
+    event = SaveEvent.create!(user_id: user.id)
+
+    event.eventable == user
+    ```
+
+    The `Event` model now has an `eventable` method which could return a `user` object
+    or an `organization` object depending on which was associated.
+    <!-- go go polymorphin power rangers -->
+    For each polymorphic association, you'll need to add a `belongs_to`. This helps to keep
+    our polymorphic associations type-safe! [See migrations](#{Guides::Database::Migrations.path(anchor: Guides::Database::Migrations::ANCHOR_ASSOCIATIONS)}) for `add_belongs_to`.
+
+    You'll also note that the `belongs_to` has nillable models. This is required for the polymorphic
+    association. Even though these are set as nilable, the association still requires at least 1 of the
+    `associations` to exist. This means that `eventable` is never actually `nil`.
+
+    If you need this association to be fully optional where `eventable` could be `nil`, you'll add the
+    `optional` option.
+
+    ```crystal
+    # eventable can now be nil
+    polymorphic eventable, optional: true, associations: [:user, :organization]
+    ```
```

Comment:
Would be good to show that you need to preload or use `!` version to skip preloading